company-keywords.el

```diff
@@ -307,7 +315,7 @@
                  (not (company-in-string-or-comment))
                  (or (company-grab-symbol) 'stop)))
     (candidates
-     (let ((completion-ignore-case nil)
+     (let ((completion-ignore-case company-keywords-ignore-case)
```

Comment:
You also need to add an `ignore-case` clause to the `cl-case` here.

See its description in the `company-backends` docstring.